plugins/org.eclipse.n4js.model/model/N4JS.xcore

```diff
@@ -1433,14 +1433,30 @@ abstract class ParameterizedAccess {
 	}
 }
 
-class ParameterizedCallExpression extends Expression, ParameterizedAccess {
-	/*
-	 * The callee or target of the call, e.g. the name of the called function. This is not the receiver (which may be null).
+/**
+ * Abstract base class for expressions that do have a target and are potentially null safe.
+ */
+abstract class ExpressionWithTarget extends Expression {
+	/**
+	 * The target of this dereferencing expression (aka receiver).
```

Comment:
Given this comment, what about renaming `ExpressionWithTarget` to `DereferencingExpression`?
Or renaming it to `AccessExpression`, considering we already have `ParameterizedAccess`?

Also, I wonder about the relation between `ExpressionWithTarget` and `ParameterizedAccess`. For example, the property `NewExpression#callee` could be seen as a "target" (new expression is similar to call expression, in that respect) and then `ParameterizedAccess` could inherit from `ExpressionWithTarget` / `AccessExpression`. Drawback: the new expression does not allow optional chaining but would then inherit property `optionalChaining`.

NOTE: this does not have to be done as part of this PR. But if not, please add this remark as a comment to the issue.